### PR TITLE
Acquire non-global lock for last acceptance deploy

### PR DESCRIPTION
**Pull request by deiwin**
This is a partial revert of ab8b6f4 ("Remove acceptance lock usage at the end
of the deploy", 2019-10-09). That commit removed all locking from that phase,
but this partially recovers that locking. The locking was removed, because the
global (`acceptance-environment`) lock was highly contended. With this new
implementation, that global lock is still not used and so the problem of
contention remains addressed. It recovers the "local" locking, however, because
as we've just found out, without it multiple deploys in the same project can
start deploying to acceptance at the same time, causing conflicts.

### CODE REVIEW

File: src/com/salemove/Deployer.groovy [hunk 1]
```diff
@@ -673,23 +675,25 @@ class Deployer implements Serializable {
...
       [resource: localLock]
     }
   }
-  private def globalLockRequired() {
+  private def globalLockRequired(args) {
     // This assumes that the arguments have already been validated
-    globalLockConfigured && getTriggerArgs(script) != Args.noGlobalLock
+    globalLockConfigured &&
+      getTriggerArgs(script) != Args.noGlobalLock &&
+      !args.onlyLocal
```
Comment (ostankin) [inline src/com/salemove/Deployer.groovy]: TBH I don't like the way this expression grows. Can you put a more comprehensive comment on what it does? So far it looks totally like some fishy magic.

Comment (deiwin) [reply src/com/salemove/Deployer.groovy]: Does https://github.com/salemove/pipeline-lib/compare/d45455ce8e7cf7a3199317fd5ea765d57aa86bfe..2d8702bcd0312520c39f35f750e180e830e1ebb3 clarify the behavior for you?

Comment (ostankin) [reply src/com/salemove/Deployer.groovy]: Yup, much better, thanks! While reviewing I spent about 20 minutes to hunt down the same info by myself :)

**Comment by deiwin** [comment]
!deploy

**Comment by sm-deployer** [comment]
@deiwin, the changes were validated in acceptance. Please click **Proceed** [here](https://ci.at.samo.io/job/salemove/job/pipeline-lib/job/PR-102/3/display/redirect) (or [in the old UI](https://ci.at.samo.io/job/salemove/job/pipeline-lib/job/PR-102/3//console)) to continue the deployment.

**Comment by deiwin** [comment]
It acquired the "local" lock as expected during the last phase:
```
Trying to acquire lock on [deploy-pipeline-test-default-acceptance-environment]
Lock acquired on [deploy-pipeline-test-default-acceptance-environment]
```

**Pull request merged**